tradebot/data/batch.py

```
@@ -0,0 +1,236 @@
+            logger.info(f"Successfully fetched {len(validated_data)} records for {symbol}")
+            return validated_data
+            
+        except Exception as e:
+            logger.error(f"Failed to fetch data for {symbol}: {e}")
+            
+            # Propagate specific exceptions without wrapping
+            if isinstance(e, InvalidSymbolError):
+                raise e
+                
+            # Wrap other exceptions in DataFetchError
+            raise DataFetchError(f"Unable to fetch data for {symbol}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)